architecture/protocol.go

```diff
@@ -147,6 +147,16 @@ func (command *Command) Parse(rawCommand string) (bool, error) {
 			return true, command.Err
 		}
 
+		// Check if parameter types are correct
+		for i, _ := range opts.Params {
+			if opts.ParamInteger[i] {
+				_, err := strconv.Atoi(parts[i+1])
```

Comment:
@vimukthi-git I thought the objective was just validating, not really parsing it only once, same as @stephenwithav said, I couldn't really find an elegant solution to have it parsed only once.

In previous commits of this PR I had it as array of interfaces to store different types in same array, maybe we could consider bringing that back?